### PR TITLE
👌 IMPROVE: MyST role syntax parsing

### PR DESCRIPTION
**Pull request by chrisjsewell**
Match characters instead of using regex.
This now allows for an unlimited name length and new lines in the content.

### CODE REVIEW

**Review by webknjaz — COMMENTED** [review]
How about adding a test case with a line break in between the backticks to cover https://github.com/executablebooks/MyST-Parser/issues/448?

File: tests/fixtures/myst_role.md [hunk 1]
```diff
@@ -3,14 +3,30 @@ Basic:
...
 
 Surrounding Text:
 .
 a {abc}`xyz` b
 .
-<p>a <code class="sphinx-role">{abc}[xyz]</code> b</p>
+<p>a <code class="myst role">{abc}[xyz]</code> b</p>
+.
+
+New lines:
```
Comment (chrisjsewell) [inline tests/fixtures/myst_role.md:23]: @webknjaz that test is already here. Is there any other fixtures you can think of before I merge

Comment (webknjaz) [reply tests/fixtures/myst_role.md:23]: How about a more complex case with spaces in the reference between angled braces?
```md
{ref}`some
text
<and
a
custom reference>`
```

Comment (chrisjsewell) [reply tests/fixtures/myst_role.md:23]: thanks for the quick reply

This complexity won't particularly make any difference here though: the markdown-plugin just stores everything inside the ticks as a raw string (now replacing new lines with spaces), then the token (with name+content) is processed specifically by myst-parser. i.e. it doesn't really matter what is inside the ticks (in the example above the token's content will be `some text <and a custom reference>`)

Comment (webknjaz) [reply tests/fixtures/myst_role.md:23]: Got it. But I'd still have some regression test for this, if not in this lib, then in some other place maybe?

Comment (chrisjsewell) [reply tests/fixtures/myst_role.md:23]: Yep, I can add a test like this in myst-parser, when I downstream this change to there 👍

**Comment by asmeurer** [comment]
Does (or could) the unlimited length thing also apply to https://github.com/executablebooks/MyST-Parser/issues/181?

**Comment by chrisjsewell** [comment]
> Does (or could) the unlimited length thing also apply to executablebooks/MyST-Parser#181?

yep indeed 👍, thank for pointing that out

**Pull request merged**